src/me/mrCookieSlime/Slimefun/listeners/FurnaceListener.java

```diff
@@ -22,24 +22,19 @@ public FurnaceListener(SlimefunStartup plugin) {
 	
 	@EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
 	public void onBurn(FurnaceBurnEvent e) {
-		if (BlockStorage.check(e.getBlock()) != null && BlockStorage.check(e.getBlock()) instanceof EnhancedFurnace) {
-			EnhancedFurnace furnace = (EnhancedFurnace) BlockStorage.check(e.getBlock());
-			if (furnace.getFuelEfficiency() > 0) e.setBurnTime(((int) ((1.0 + 0.2 * furnace.getFuelEfficiency()) * e.getBurnTime())));
-		}
+		EnhancedFurnace furnace = (EnhancedFurnace) BlockStorage.check(e.getBlock());
+		if (furnace instanceof EnhancedFurnace && furnace.getFuelEfficiency() > 0)
```

Comment:
Never do an "instanceof"-Check **after** Casting.
This would result in a ClassCastException